packages/angular_devkit/build_angular/src/differential-loading/utils.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as browserslist from 'browserslist';
+import * as caniuse from 'caniuse-api';
+import * as path from 'path';
+import * as ts from 'typescript';
+
+
+export function isDifferentialLoadingNeeded(
+  tsConfigFileName: string, browserslistQuery: string[] | undefined = undefined): boolean {
+  const es5SupportNeeded = isEs5BrowserSupportNeeded(tsConfigFileName, browserslistQuery);
+  const target = getTargetVersion(tsConfigFileName);
+
+  return es5SupportNeeded && target !== ts.ScriptTarget.ES5;
+}
+
+export function getTargetVersion(tsConfigFileName: string): ts.ScriptTarget {
+
+  const config = ts.getParsedCommandLineOfConfigFile(
+    tsConfigFileName,
+    {},
+    { ...ts.sys, onUnRecoverableConfigFileDiagnostic: () => {} },
+  );
+
+  if (typeof config === 'undefined') {
+    throw new Error('could not read tsconfig: ' + tsConfigFileName);
+  }
+
+  const compilerOptions = config.options;
+  const target = compilerOptions.target;
+
+  if (typeof target === 'undefined') {
+
+    // fall back to ES5
+    return ts.ScriptTarget.ES5;
+  }
+
+  return target;
+}
+
+export function isEs5BrowserSupportNeeded(
+  tsConfigFileName: string | null, browserslistQuery: string[] | undefined = undefined): boolean {
```

Comment:
Wouldn't these browser list be coming from `browserslist` file which is being read below?

Reply:
Absolutely. I'm currently using the option to pass in a `browserslistQuery` instead of reading the `browserslist` file for testing.

Reply:
Ummm, probably we need to remove these before merging this PR.

Reply:
I've changed it now. The unit tests read the browserlistQuerys now from the hard disk. Hence, there is no need for this `browserslistQuery` parameter anymore.